### DISCUSSION TITLE
Link syntax question

### DISCUSSION THREAD

**Discussion by sarmong** [nvim-neorg/neorg#421] [Syntax] [answered]
Is there a specific reason for why link syntax has URL first and title last?

It is not very convenient to look at the very end of the line searching for title.

I think title is much more important than the URL and should come first.

**Comment by katawful** [comment]
Ideally the entire link format will be concealled fully like you might find in emacs w/ orgmode, but that requires a bit more work on Neovim's end to be seamless. Current concealing should allow for it to work, but probably doesn't work perfectly like 1-2 character conceals

Title comes after because it's *modifying* the link. It is consistent across other things like tags in which the object is defined first then modified

**Reply by max397574** [reply]
Concealing work great if you're on nightly
![Screenshot 2022-04-05 at 17 20 51](https://user-images.githubusercontent.com/81827001/161788200-d4023bd1-8a1a-485c-82b5-1a8f3d10ed69.png)
![Screenshot 2022-04-05 at 17 20 58](https://user-images.githubusercontent.com/81827001/161788225-a59a4eb0-b1a4-4f8b-b961-d219c9c13518.png)

**Reply by sarmong** [reply]
Yes, it would be great with concealing, but I think it should be usable without it too. 
And with the current order it is not really usable. IMHO, markdown does it better

**Comment by vhyrro** [comment, marked as answer]
Heyo, descriptions can actually be placed in front of the link, like this: `[description]{# link}`. Do note that this syntax is an *anchor* versus a regular plain link. Anchors are the same thing as links when used standalone, so you can interoperate them as you see fit. Anchors just add an extra level of optional features:

```
Check out this [blog].

This is my [blog]{https://my-blog.example.com}.
```

The first line's `[blog]` link will directly link to `https://my-blog.example.com`, because Neorg will see that you've defined the location for `[blog]` lower down below. This isn't a feature you need to use though, so `[title]{link}` is completely valid on its own. Check [here](https://github.com/nvim-neorg/neorg/discussions/336#discussioncomment-2045519) for more info if you want :)

**Reply by sarmong** [reply]
Thanks, that's great!
